### PR TITLE
Storage containers splitting

### PR DESCRIPTION
**Pull request by artek-koltun**
Provide an ability to run the images as
standalone containers outside the test scope.

### CODE REVIEW

File: .gitmodules
Comment (mestery) [inline .gitmodules:1]: It feels like we should update a README somewhere to make sure we note you now to clone the repository with `--recursive` here, but I see our READMEs are a bit of a mess. I filed #156 to track this.

File: build/scripts/run-shellcheck.sh
Comment (mestery) [inline build/scripts/run-shellcheck.sh:15]: Hmmmm .... you're skipping shellcheck here? I think instead we need to fix the static analysis failures.

File: build/storage/Dockerfile
Comment (mestery) [inline build/storage/Dockerfile:13]: Fedora 33 is super ancient. Given that #134 is already open, I think this should be updated to use Fedora 36.

**Review by mestery — CHANGES_REQUESTED** [review]
This is a *massive* PR. I have taken a quick look and proposed some required things which need to happen before it can be merged. One other thing here, I don't know how you made the diagrams here, but checking in .svg files is challenging because you can't edit them easily. It would be better to use [draw.io](https://app.diagrams.net) to make the graphics and save them as editable .png files. Ping me if you need help here.

File: .gitmodules
Comment (stolsma) [reply .gitmodules:1]: Yeah, do we want to use submodules or do we keep using git clone/downloads in scripts? I prefer the latter but possibly I don't know the positive points of using submodules...

Comment (intelfisz) [reply .gitmodules:1]: There are opponents and proponents of both approaches. We had the guideline to include SPDK as a submodule with IPDK.

File: build/scripts/run-shellcheck.sh
Comment (intelfisz) [reply build/scripts/run-shellcheck.sh:15]: Hmmm... I don't think we want to check and fix potential failures from 3rd party projects - this excludes spdk folder from the check.

File: build/storage/core/host-target/device_exerciser.py
Comment (artek-koltun) [reply build/storage/core/host-target/device_exerciser.py:1]: @mestery do I need to use the default formatting rules or we have any specific ones?

File: build/storage/Dockerfile
Comment (artek-koltun) [reply build/storage/Dockerfile]: Updated to fedora:36

File: build/scripts/run-shellcheck.sh
Comment (mestery) [reply build/scripts/run-shellcheck.sh:15]: I guess someone should file an issue with SPDK to fix their static analysis failures in their bash scripts then. :)

File: build/storage/core/host-target/device_exerciser.py
Comment (mestery) [reply build/storage/core/host-target/device_exerciser.py:1]: I will leave that up to you and the tam @artek-koltun on which formatting rules to apply.

File: .gitmodules
Comment (mestery) [reply .gitmodules:1]: Lets leave the submodule in here for now to move this forward. I believe as we look to figure out a better long-term strategy for the repositories which make up IPDK, we can figure out if the submodule is necessary or not.

File: build/storage/Dockerfile
Comment (mestery) [reply build/storage/Dockerfile]: Thanks!

File: build/storage/core/host-target/device_exerciser.py
Comment (mestery) [reply build/storage/core/host-target/device_exerciser.py:1]: Thank you!

**Review by mestery — APPROVED** [review]
Thanks or addressing my concerns!

**Pull request merged**
